### DISCUSSION TITLE
Float numbers as edge labels in link prediction

### DISCUSSION THREAD

**Discussion by faren-f** [pyg-team/pytorch_geometric#4310] [Q&A]
I am using PyG for link prediction on a heterogeneous graph based on [this example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/hetero_link_pred.py). There, the edge labels are integer numbers from 1 to 5; but, in my data, edge labels are float numbers from 0 to 4, and my task is to predict continuous edge labels by reducing the MSE. When I use RandomLinkSplit for cross-validation, I am facing an error at RandomLinkSplit-line 267 that says I cannot have float numbers as edge labels:

```
assert edge_label.dtype == torch.long   
```

Since I do not need negative sampling here, I tried removing the command above. The algorithm worked but the model does not seem to be trained properly and the results are not good. I really appreciate it if you would comment on my problem to see if I am allowed to remove this command or there are other solutions for my task.

**Comment by rusty1s** [comment]
Thanks for reporting. You can safely remove it (as long as `neg_sampling_ratio == 0`) and I fixed it in https://github.com/pyg-team/pytorch_geometric/pull/4311. Our link pred example works in a similar fashion as well (it is also a regression dataset), but did not crash since labels were stored as `torch.long` :)

If your model is underperforming, I feel that there needs to exist another issue.

**Reply by faren-f** [reply]
Thank you so much for your quick response!
Now I am sure about this part and I can continue with other parts :)
